libfreerdp/core/settings.c

```diff
@@ -277,6 +277,199 @@ void freerdp_settings_print_warnings(const rdpSettings* settings)
 	}
 }
 
+static BOOL monitor_operlaps(const rdpSettings* settings, UINT32 start, UINT32 count,
+                             const rdpMonitor* compare)
+{
+	const RECTANGLE_16 rect1 = { .left = compare->x,
+		                         .top = compare->y,
+		                         .right = compare->x + compare->width,
+		                         .bottom = compare->y + compare->height };
+	for (UINT32 x = start; x < count; x++)
+	{
+		const rdpMonitor* monitor =
+		    freerdp_settings_get_pointer_array(settings, FreeRDP_MonitorDefArray, x);
+		const RECTANGLE_16 rect2 = { .left = monitor->x,
+			                         .top = monitor->y,
+			                         .right = monitor->x + monitor->width,
+			                         .bottom = monitor->y + monitor->height };
+
+		if (rectangles_intersects(&rect1, &rect2))
+		{
+			WLog_ERR(
+			    TAG,
+			    "Mulitimonitor mode requested, but local layout has gaps or overlapping areas!");
+			WLog_ERR(TAG, "Please reconfigure your local monitor setup so that thre are no gaps or "
+			              "overlapping areas!");
+			return TRUE;
+		}
+	}
+	return FALSE;
+}
+
+static BOOL monitor_has_gaps(const rdpSettings* settings, UINT32 start, UINT32 count,
+                             const rdpMonitor* compare)
+{
+	BOOL hasNeighbor = FALSE;
+	for (UINT32 x = 0; x < count; x++)
+	{
+		if (x == start)
+			continue;
+
+		const rdpMonitor* monitor =
+		    freerdp_settings_get_pointer_array(settings, FreeRDP_MonitorDefArray, x);
+
+		if (compare->x == monitor->x + monitor->width)
+			hasNeighbor = TRUE;
+		if (compare->y == monitor->y + monitor->height)
+			hasNeighbor = TRUE;
+		if (compare->x + compare->width == monitor->x)
+			hasNeighbor = TRUE;
+		if (compare->y + compare->height == monitor->y)
+			hasNeighbor = TRUE;
+	}
+	if (!hasNeighbor)
+	{
+		WLog_ERR(TAG,
+		         "Monitor configuration has gaps! Monitor %" PRIu32 " does not have any neighbor",
+		         start);
+	}
+	return !hasNeighbor;
+}
+
+static BOOL freerdp_settings_client_monitors_have_gaps(const rdpSettings* settings)
```

Comment:
I may be wrong but to check that you can also add the rects of monitors to a `REGION16` and look at the intersection with its bounding rect, if the result is the same region, you don't have gaps.

Reply:
@hardening implemented now with:
1. a function `align_rects` which checks that either left, top, bottom or right of two monitors are are aligned
2. `dijkstra` to build a graph from monitor 0 and check all others are reachable.